typing/cmi_format.ml

```diff
@@ -20,8 +20,7 @@ type pers_flags =
   | Unsafe_string
 
 type error =
-    Not_an_interface of string
-  | Wrong_version_interface of string * string
+    Not_an_interface of string * string option
```

Comment:
I did consider this -- and the `parse_error` structure has grown closed to these module-specific `error` structures over time (in particular I inherited the Corrupted/Not_a_magic_number distinction from them), but I think that this way is better:

- it is more consistent with other exception constructors that tend to often carry "unstructured" rather than "structured" payloads (I'm not for keeping bad practices indefinitely by "coherence in abstraction levels" is good as well)
- this allows me to isolate the `Magic_number`-using code in only one part of the module, the one reading the files. Otherwise I would have `Magic_number` usage spread to both the exception-raisers and exception-catchers.